Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteGPUProxy.cpp

```diff
@@ -43,64 +43,41 @@
     if (!clientConnection)
         return nullptr;
-    auto remoteGPUProxy = adoptRef(new RemoteGPUProxy(gpuProcessConnection, clientConnection.releaseNonNull(), convertToBackingContext, identifier));
+    auto remoteGPUProxy = adoptRef(new RemoteGPUProxy(remoteRenderingBackendProxy, clientConnection.releaseNonNull(), convertToBackingContext, identifier));
     remoteGPUProxy->initializeIPC(WTFMove(serverConnectionHandle), renderingBackend);
     return remoteGPUProxy;
 }
 
 
-RemoteGPUProxy::RemoteGPUProxy(GPUProcessConnection& gpuProcessConnection, Ref<IPC::StreamClientConnection> clientConnection, WebGPU::ConvertToBackingContext& convertToBackingContext, WebGPUIdentifier identifier)
+RemoteGPUProxy::RemoteGPUProxy(RemoteRenderingBackendProxy& remoteRenderingBackendProxy, Ref<IPC::StreamClientConnection> clientConnection, WebGPU::ConvertToBackingContext& convertToBackingContext, WebGPUIdentifier identifier)
     : m_backing(identifier)
     , m_convertToBackingContext(convertToBackingContext)
-    , m_gpuProcessConnection(&gpuProcessConnection)
+    , m_remoteRenderingBackendProxy(remoteRenderingBackendProxy)
     , m_streamConnection(WTFMove(clientConnection))
 {
 }
 
 RemoteGPUProxy::~RemoteGPUProxy()
 {
-    disconnectGpuProcessIfNeeded();
+    m_streamConnection->invalidate();
+    m_remoteRenderingBackendProxy.streamConnection().send(Messages::RemoteRenderingBackend::ReleaseRemoteGPU(m_backing), m_remoteRenderingBackendProxy.renderingBackendIdentifier(), Seconds::infinity());
 }
 
 void RemoteGPUProxy::initializeIPC(IPC::StreamServerConnection::Handle&& serverConnectionHandle, RenderingBackendIdentifier renderingBackend)
 {
-    m_gpuProcessConnection->addClient(*this);
-    m_gpuProcessConnection->connection().send(Messages::GPUConnectionToWebProcess::CreateRemoteGPU(m_backing, renderingBackend, WTFMove(serverConnectionHandle)), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
+    m_remoteRenderingBackendProxy.streamConnection().send(Messages::RemoteRenderingBackend::CreateRemoteGPU(m_backing, WTFMove(serverConnectionHandle)), m_remoteRenderingBackendProxy.renderingBackendIdentifier(), Seconds::infinity());
     m_streamConnection->open(*this);
     // TODO: We must wait until initialized, because at the moment we cannot receive IPC messages
     // during wait while in synchronous stream send. Should be fixed as part of https://bugs.webkit.org/show_bug.cgi?id=217211.
     waitUntilInitialized();
 }
 
-void RemoteGPUProxy::disconnectGpuProcessIfNeeded()
-{
-    if (m_gpuProcessConnection) {
-        m_streamConnection->invalidate();
-        m_gpuProcessConnection->connection().send(Messages::GPUConnectionToWebProcess::ReleaseRemoteGPU(m_backing), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
-        m_gpuProcessConnection = nullptr;
-    }
-}
-
-void RemoteGPUProxy::gpuProcessConnectionDidClose(GPUProcessConnection& connection)
```

Comment:
This could be an implementation of `didClose` (which currently has an empty impl in the header), so that we still get notifications when the GPU crashes.

I'm not sure m_lost is adding a lot of value right now, but we should either continue to set it, or get rid of it.